src/azul/__init__.py

```diff
@@ -1165,6 +1167,20 @@ def is_lower_sandbox(self) -> bool:
         def is_unit_test(self):
             return self.name == self.test_name
 
+        @cached_property
```

Comment:
The overhead of caching is probably more expensive than the actual method call here.
```suggestion
        @property
```